### PR TITLE
v1 draft, NG approval variables CDS-74197

### PR DESCRIPTION
**Pull request by douglas-j-bothwell**
# Harness Developer Pull Request

https://harness.atlassian.net/browse/CDS-74197

**Main topic:** Built-in and custom Harness variables reference > [Approval](https://651593d7d7e9ca131a23b168--harness-developer.netlify.app/docs/platform/variables-and-expressions/harness-variables#approval)

New sections that link to main topic:
- Using manual Harness approval steps in CD stages [Approval variables](https://651593d7d7e9ca131a23b168--harness-developer.netlify.app/docs/continuous-delivery/x-platform-cd-features/cd-steps/approvals/using-harness-approval-steps-in-cd-stages/#approval-variables)
- Jira approval steps [Approval variables](https://651593d7d7e9ca131a23b168--harness-developer.netlify.app/docs/continuous-delivery/x-platform-cd-features/cd-steps/approvals/using-jira-and-service-now-approval-steps-in-cd-stages#approval-variables)
- ServiceNow approval steps [Approval variables](https://651593d7d7e9ca131a23b168--harness-developer.netlify.app/docs/continuous-delivery/x-platform-cd-features/cd-steps/approvals/using-jira-and-service-now-approval-steps-in-cd-stages#approval-variables)

### CODE REVIEW

File: docs/continuous-delivery/shared/servicenow-approval-steps.md [hunk 1]
```diff
@@ -115,6 +115,10 @@ For example, `<+ticket.state.displayValue> == "New"` in the Approval Criteria,
 
 For details, go to **Custom table support** in [Create ServiceNow tickets in CD stages](/docs/continuous-delivery/x-platform-cd-features/cd-steps/ticketing-systems/create-service-now-tickets-in-cd-stages#custom-table-support).
 
+## Approval variables
+
+After an approval is granted, [<+approval>](/docs/platform/variables-and-expressions/harness-variables#approval) variables store the approver name and email as well as any approval comments. These variables are useful if you want to the pipeline to generate notifications about the approval.
```
Comment (douglas-j-bothwell) [inline docs/continuous-delivery/shared/servicenow-approval-steps.md:120]:
```suggestion
After an approval is granted, [<+approval>](/docs/platform/variables-and-expressions/harness-variables#approval) variables store the approver name and email as well as any approval comments. These variables are useful if you want the pipeline to generate notifications about the approval.
```

File: docs/continuous-delivery/x-platform-cd-features/cd-steps/approvals/using-harness-approval-steps-in-cd-stages.md [hunk 2]
```diff
@@ -167,6 +167,10 @@ In **Advanced**, you can use the following options:
 * [Looping Strategy](/docs/platform/pipelines/looping-strategies-matrix-repeat-and-parallelism/)
 * [Policy Enforcement](/docs/platform/governance/Policy-as-code/harness-governance-overview)
 
+## Approval variables
+
+After an approval is granted, [<+approval>](/docs/platform/variables-and-expressions/harness-variables#approval) variables store the approver name and email as well as any approval comments. These variables are useful if you want to the pipeline to generate notifications about the approval.
```
Comment (douglas-j-bothwell) [inline docs/continuous-delivery/x-platform-cd-features/cd-steps/approvals/using-harness-approval-steps-in-cd-stages.md:172]:
```suggestion
After an approval is granted, [<+approval>](/docs/platform/variables-and-expressions/harness-variables#approval) variables store the approver name and email as well as any approval comments. These variables are useful if you want the pipeline to generate notifications about the approval.
```

**Review by douglas-j-bothwell — COMMENTED** [review]
Minor tweaks

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 3]
```diff
@@ -1260,6 +1260,30 @@ Here are the sidecar expressions:
 * `<+artifacts.sidecars.SIDECAR_IDENTIFIER.tag>`
 * `<+artifacts.sidecars.SIDECAR_IDENTIFIER.connectorRef>`
 
+## Approval
+
+Whenever a user grants an approval in an Approval step, the pipeline maintains the user information of the approver for the rest of the pipeline execution.  You can use these variables in notifications after an approval gets granted.
```
Comment (SudheendraKatte) [inline docs/platform/variables-and-expressions/harness-variables.md:1265]:
```suggestion
Whenever a user grants an approval in an Approval step, the pipeline maintains the user information of the approver for the rest of the pipeline execution. You can use these variables in notifications after an approval is granted.  
```

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 4]
```diff
@@ -1260,6 +1260,30 @@ Here are the sidecar expressions:
...
+
+The Harness username of the approver.
+
+### <+approval.approvalActivities[0].user.email>
+
+The email address of the approver.
+
+### <+approval.approvalActivities[0].comments>
+
+Any comments the user added to the approval, formatted as a single string. This variable is populated from the Comment output variable in the approval step or stage.
```
Comment (SudheendraKatte) [inline docs/platform/variables-and-expressions/harness-variables.md:1283]: How about this:
User comments from the approval process are now consolidated into a single formatted string. This string is populated using the 'Comment' output variable from the approval step or stage.

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 5]
```diff
@@ -1260,6 +1260,30 @@ Here are the sidecar expressions:
 * `<+artifacts.sidecars.SIDECAR_IDENTIFIER.tag>`
 * `<+artifacts.sidecars.SIDECAR_IDENTIFIER.connectorRef>`
 
+## Approval
+
+Whenever a user grants an approval in an Approval step, the pipeline maintains the user information of the approver for the rest of the pipeline execution.  You can use these variables in notifications after an approval gets granted.  
+
+These variables are available for Approval steps only, not stages. 
+
+In the following example, a Deploy stage has two Approval steps. For each approval, the pipeline maintains a separate set of approval variables. Use the index to access the array element that contains the variables for the relevant approval.
```
Comment (SudheendraKatte) [inline docs/platform/variables-and-expressions/harness-variables.md:1269]: How about this:
To access the variables associated with a specific approval, use the index to reference the array element containing those variables.

**Review by SudheendraKatte — APPROVED** [review]
Approved with minor suggestions.

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 6]
```diff
@@ -1260,6 +1260,30 @@ Here are the sidecar expressions:
 * `<+artifacts.sidecars.SIDECAR_IDENTIFIER.tag>`
 * `<+artifacts.sidecars.SIDECAR_IDENTIFIER.connectorRef>`
 
+## Approval
+
+Whenever a user grants an approval in an Approval step, the pipeline maintains the user information of the approver for the rest of the pipeline execution. You can use these variables in notifications after an approval is granted.  
+
+These variables are available for Approval steps only, not stages. 
+
+In the following example, a Deploy stage has two Approval steps. For each approval, the pipeline maintains a separate set of approval variables. Use the index to access the array element that contains the variables for the relevant approval.
```
Comment (douglas-j-bothwell) [inline docs/platform/variables-and-expressions/harness-variables.md:1269]:
```suggestion
In the following example, a Deploy stage has two Approval steps. For each approval, the pipeline maintains a separate set of approval variables. Use the array index to access the variables for a specific approval.
```

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 7]
```diff
@@ -1260,6 +1260,30 @@ Here are the sidecar expressions:
...
+
+The Harness username of the approver.
+
+### <+approval.approvalActivities[0].user.email>
+
+The email address of the approver.
+
+### <+approval.approvalActivities[0].comments>
+
+Any comments the user added to the approval, formatted as a single string. This variable is populated from the Comment output variable in the approval step or stage.
```
Comment (douglas-j-bothwell) [inline docs/platform/variables-and-expressions/harness-variables.md:1283]:
```suggestion
User comments from the approval, formatted as a single string. This variable is populated from the Comment output variable generated by the approval step.  
```

**Pull request merged**